tests/Unit/Adapter/Image/ProductImageFileValidatorTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter\Image;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Image\ProductImageFileValidator;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageUploadException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedImageConstraintException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedImageSizeException;
+use Tests\Resources\DummyFileUploader;
+
+class ProductImageFileValidatorTest extends TestCase
+{
+    /**
+     * @dataProvider getInvalidMaxUploadSizesForFile
+     *
+     * @param string $filePath
+     * @param int $maxPhpIniUploadSize
+     * @param int $maxUploadQuotaSize
+     *
+     * @throws UploadedImageConstraintException
+     * @throws ImageUploadException
+     */
+    public function testItThrowsExceptionWhenFileSizeIsLargerThanMaxUploadSize(
```

Comment:
Exception cases are correctly tested, but you should also test some use cases that don't throw exceptions Just to make sure the validator doesn't always throw an exception 😅

Reply:
Also you didn't test this part `ImageManager::checkImageMemoryLimit`